lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,189 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+static struct uk_falloc *generate(void *fa_meta, void *start_zone, unsigned long len,
+									__paddr_t *new_start, __sz fa_meta_size, unsigned long *pages)
+{
+	__vaddr_t dm_off;
+	__paddr_t start = (__paddr_t)start_zone;
+	struct uk_falloc *fa = malloc(uk_fallocbuddy_size());
+
+	uk_fallocbuddy_init(fa);
+	start = PAGE_ALIGN_UP(start + fa_meta_size);
+	*pages = (len - fa_meta_size) >> PAGE_SHIFT;
+	dm_off = start;
+	fa->addmem(fa, fa_meta, start, *pages, dm_off);
+	*new_start = start;
+	return fa;
+}
+
+
+UK_TESTCASE(ukfallocbuddy, test_init)
+{
+	// test if the init function works
+
+	struct uk_falloc *fa;
+
+	fa = malloc(uk_fallocbuddy_size());
+	UK_TEST_EXPECT_SNUM_EQ(uk_fallocbuddy_init(fa), 0);
+	free(fa);
+}
+
+
+UK_TESTCASE(ukfallocbuddy, test_add_memory)
+{
+	// test if the addmem functionality works
+
+	__sz len = 10000;
+	void *start_zone = malloc(len);
+	__vaddr_t dm_off;
+	__paddr_t start = (__paddr_t)start_zone;
+	unsigned long pages = len >> PAGE_SHIFT;
+	__sz fa_meta_size = uk_fallocbuddy_metadata_size(pages);
+	void *fa_meta = malloc(fa_meta_size);
+	struct uk_falloc *fa = malloc(uk_fallocbuddy_size());
+	int addmem_result;
+
+	uk_fallocbuddy_init(fa);
+
+	start = PAGE_ALIGN_UP(start + fa_meta_size);
+	pages = (len - fa_meta_size) >> PAGE_SHIFT;
+	dm_off = start;
+	addmem_result = fa->addmem(fa, fa_meta, start, pages, dm_off);
+	UK_TEST_EXPECT_SNUM_EQ(addmem_result, 0);
+	free(start_zone);
+	free(fa_meta);
+	free(fa);
+}
+
+UK_TESTCASE(ukfallocbuddy, test_check_variables)
+{
+	__paddr_t start;
+	__sz len = 10000;
+	unsigned long pages = len >> PAGE_SHIFT;
+	__sz fa_meta_size =  uk_fallocbuddy_metadata_size(pages);
```

Comment:
```suggestion
	__sz fa_meta_size = uk_fallocbuddy_metadata_size(pages);
```
There is an extra space here